packages/edit-post/src/components/header/template-title/index.js

```diff
@@ -33,12 +41,33 @@ function TemplateTitle() {
 	}
 
 	return (
-		<span className="edit-post-template-title">
-			{
-				/* translators: 1: Template name. */
-				sprintf( __( 'Editing template: %s' ), templateTitle )
-			}
-		</span>
+		<Dropdown
+			position="bottom center"
+			className="edit-post-template-top-area"
+			contentClassName="edit-post-template-top-area__popover"
+			renderToggle={ ( { onToggle } ) => (
+				<>
+					<div className="edit-post-template-title">
+						{ __( 'About' ) }
+					</div>
+					<Button isSmall isTertiary onClick={ onToggle }>
+						{ templateTitle }
+					</Button>
+				</>
+			) }
+			renderContent={ () => (
+				<NavigableMenu>
```

Comment:
If we want to achieve something like the example:
![image](https://user-images.githubusercontent.com/11271197/117725548-30d3ae80-b1dd-11eb-92f9-06dc50e6257a.png)

I think it's probably better to avoid using a NavigableMenu/MenuGroup/MenuItem, which should generally only be used for vertical list menus. Those components are not needed in a plain dropdown and will make it difficult to achieve the desired result. I think the screenshot above is closer to something like the Link UI in terms of the way it's presented. Arrow key navigation isn't needed for that kind of component, tabs are fine.

An example is here where the Delete button takes up the full width because it's a menu item:
<img width="375" alt="Screenshot 2021-05-19 at 5 44 43 pm" src="https://user-images.githubusercontent.com/677833/118793212-f6e46580-b8ca-11eb-8156-3e1d22d3d35b.png">

Reply:
Although I see in the issue that this is the latest quick win design:
<img width="411" alt="Screenshot 2021-04-22 at 12 17 03" src="https://user-images.githubusercontent.com/846565/115705633-b1348b80-a364-11eb-8bff-f84c82c22f08.png">

I still think it'd be better not to use NavigableMenu because there's only one button that can be tabbed to.

Reply:
A close button needs to be added because it is now a keyboard trap.

Reply:
Oh, there are some other designs proposed in this PR as well, not sure which is the latest 😄 

> A close button needs to be added because it is now a keyboard trap.

Escape or clicking outside should be fine, most popovers in the editor do this.